Assets/Scripts/Tools/Extrude.cs

```diff
@@ -70,37 +58,59 @@ public override void PrimaryAction()
         MoveFace moveFace = faceObj.gameObject.GetComponent<MoveFace>();
         MeshRebuilder meshRebuilder = moveFace.meshRebuilder;
         Mesh mesh = moveFace.mesh;
+        distance = 1f;
         extrudeFace(faceObj.id, meshRebuilder, mesh);
     }
 
-    // Change material, disable vertex grab interactable, set boolean
+
+    // Extrude and move on first press, stop moving on second press
+    public override void SecondaryAction()
+    {
+        if(movingNewFace)
+        {
+            print("test stop interaction");
+            directInteractor.EndManualInteraction();
+            movingNewFace = false;
+            return;
+        }
+       
+        if(currentFace == null)
+            return;
+
+        Face faceObj = currentFace.GetComponent<Face>();
+        MoveFace moveFace = faceObj.gameObject.GetComponent<MoveFace>();
+        MeshRebuilder meshRebuilder = moveFace.meshRebuilder;
+        Mesh mesh = moveFace.mesh;
+
+        
+
+        if(inRadius && !movingNewFace)
+        {
+            distance = 5f;
+            extrudeFace(faceObj.id, meshRebuilder, mesh);
+            moveNewFace(meshRebuilder);
+            return;
+        }        
+    }
+
+    // Extrude face along normal by distance value set by either button
     public void extrudeFace(int faceId, MeshRebuilder meshRebuilder, Mesh mesh, bool sendFaceExtrudeEvent = true)
     {
         Face faceObj = meshRebuilder.faceObjects[faceId];
         Vertex vertex1 = meshRebuilder.vertexObjects[faceObj.vert1];
         Vertex vertex2 = meshRebuilder.vertexObjects[faceObj.vert2];
         Vertex vertex3 = meshRebuilder.vertexObjects[faceObj.vert3];
 
-        // Don't extrude if one of the vertices is locked
-        // if (vertex1.GetComponent<MoveVertices>().isLocked ||
-        //     vertex2.GetComponent<MoveVertices>().isLocked ||
-        //     vertex3.GetComponent<MoveVertices>().isLocked)
-        // {
-        //     print("Unlock face vertices before extruding.");
-        //     return;
-        // }
-
         // calculating normals is done in mesh rebuilder
 
         // align new vertices w/ face normal
-        Vector3 new1 = vertex1.transform.localPosition + ((faceObj.normal  + vertex1.transform.localPosition *.005f).normalized );
-        Vector3 new2 = vertex2.transform.localPosition + ((faceObj.normal  + vertex2.transform.localPosition *.005f).normalized );
-        Vector3 new3 = vertex3.transform.localPosition + ((faceObj.normal  + vertex3.transform.localPosition *.005f).normalized );
+        Vector3 new1 = vertex1.transform.localPosition + ((faceObj.normal  + vertex1.transform.localPosition *.005f).normalized ) / distance;
```

Comment:
i think we are dividing by zero here in the case where the extrude event is sent to the other players

Reply:
one way to fix this error is to send the distance in the event